Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1352 @@
+                    _fills.TryRemove(order.Id, out ignored);
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, OrderFee.Zero, "Zerodha Order Rejected Event: " + orderUpdate.StatusMessage) { Status = OrderStatus.Canceled });
+                }
+
+                if (orderUpdate.FilledQuantity > 0)
+                {
+                    var symbol = _symbolMapper.ConvertZerodhaSymbolToLeanSymbol(orderUpdate.InstrumentToken);
+                    var fillPrice = orderUpdate.AveragePrice;
+                    var fillQuantity = orderUpdate.FilledQuantity;
+                    var direction = orderUpdate.TransactionType == "SELL" ? OrderDirection.Sell : OrderDirection.Buy;
+                    var updTime = orderUpdate.OrderTimestamp.GetValueOrDefault();
+                    var orderFee = new OrderFee(new CashAmount(
+                                CalculateBrokerageOrderFee(orderUpdate.AveragePrice * orderUpdate.FilledQuantity),
+                            Currencies.INR
+                        ));
+                    if (direction == OrderDirection.Sell)
+                    {
+                        fillQuantity = -1 * fillQuantity;
+                    }
+                    var status = OrderStatus.Filled;
+                    if (fillQuantity != order.Quantity)
+                    {
+                        decimal totalFillQuantity;
+                        _fills.TryGetValue(order.Id, out totalFillQuantity);
+                        totalFillQuantity += fillQuantity;
+                        _fills[order.Id] = totalFillQuantity;
+                        status = totalFillQuantity == order.Quantity
+                            ? OrderStatus.Filled
+                            : OrderStatus.PartiallyFilled;
+                    }
+
+                    var orderEvent = new OrderEvent
+                    (
+                        order.Id, symbol, updTime, status,
+                        direction, fillPrice, fillQuantity,
+                        orderFee, $"Zerodha Order Event {direction}"
+                    );
+
+                    // if the order is closed, we no longer need it in the active order list
+                    if (status == OrderStatus.Filled)
+                    {
+                        Order outOrder;
+                        CachedOrderIDs.TryRemove(order.Id, out outOrder);
+                        decimal ignored;
+                        _fills.TryRemove(order.Id, out ignored);
+                    }
+
+                    OnOrderEvent(orderEvent);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error(e);
+                throw;
+            }
+        }
+        /// <summary>
+        /// Zerodha brokerage fees calculation
+        /// </summary>
+        /// <param name="orderValue"></param>
+        /// <returns></returns>
+        private decimal CalculateBrokerageOrderFee(decimal orderValue)
```

Comment:
This logic is in the `ZerodhaFeeModel`, wonder if maybe we should directly call the security.FeeModel?